packages/features/bookings/Booker/components/EventMeta.tsx

```diff
@@ -44,6 +46,20 @@ export const EventMeta = ({
   const isEmbed = useIsEmbed();
   const hideEventTypeDetails = isEmbed ? embedUiConfig.hideEventTypeDetails : false;
 
+  const { data: eventOwnerDefaultShedule } = trpc.viewer.availability.schedule.get.useQuery({
+    scheduleId: event?.owner?.defaultScheduleId || undefined,
+  });
+
+  useEffect(() => {
+    if (event && event?.lockTimeZoneToggleOnBookingPage) {
+      if (event?.schedule?.timeZone) {
+        setTimezone(event.schedule?.timeZone);
+      } else if (eventOwnerDefaultShedule) {
+        setTimezone(eventOwnerDefaultShedule.timeZone);
+      }
+    }
+  }, [event, setTimezone, eventOwnerDefaultShedule]);
```

Comment:
**Blocking**: Let's move the logic of getting the owner's timezone to server side